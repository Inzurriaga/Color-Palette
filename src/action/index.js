export const toggleProjectDisplay  = () => ({
    type: "TOGGLE_PROJECT"
})

export const generateColorPalette = (palette) => ({
    type: "GENERATED_PALETTE",
    palette
})

export const toggleSaveModal = () => ({
    type: "TOGGLE_SAVE"
})

export const initialProjects = (projects) => ({
    type: "INITIAL_PROJECTS",
    projects
})

export const addNewProject = (project) => ({
    type: "ADD_PROJECT",
    project
})

export const initialPalette = (palettes) =>({
    type: "INITIAL_PALETTE",
    palettes
})

export const addNewPalette = (palette) => ({
    type: "ADD_PALETTE", 
    palette
})

export const deleteProject = (id) => ({
    type: "DELETE_PROJECT",
    id
})

export const deletePalette = (id) => ({
    type: "DELETE_Palette",
    id
})

export const deleteProjectPalettes = (projectId) => ({
    type: "DELETE_PROJECT_PALETTE",
    projectId
})