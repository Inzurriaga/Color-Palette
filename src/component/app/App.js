import React from 'react';
import { connect } from "react-redux"
import Header from "../header/Header.js"
import ColorPalette from "../colorPalette/ColorPalette.js"
import ProjectDisplay from "../projectDisplay/ProjectDisplay.js"
import SaveModal from "../modal/saveModal"

export function App(props) {
  return (
    <div className="App">
      <Header/>
      <ColorPalette/>
      <ProjectDisplay/>
      {
          props.toggleSaveModal && <SaveModal />
      }
    </div>
  );
}

export const mapStateToProps = (state) => ({
  toggleSaveModal: state.toggleSaveModal
}) 

export default connect(mapStateToProps)(App);
