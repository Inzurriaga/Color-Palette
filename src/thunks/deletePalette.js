import { deletePaletteInState } from "../action"

export const deletePalette = (id) => {
    const method = {
        method: "DELETE",
    }
    return async (dispatch) => {
        try{
            const response = await fetch(process.env.REACT_APP_BACKEND_URL + `/api/${id}`, method)
            if(!response.ok) {
                throw Error(response.statusText)
              }
            dispatch(deletePaletteInState(id))
        }catch(error){

        }
    }
}