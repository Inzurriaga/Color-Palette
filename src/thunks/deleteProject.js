import { deleteProjectInState, deleteProjectPalettes } from "../action"

export const deleteProject = (id) => {
    const method = {
        method: "DELETE",
    }
    return async (dispatch) => {
        try{
            const response = await fetch(`http://localhost:3001/api/projects/${id}`, method)
            dispatch(deleteProjectInState(id))
            dispatch(deleteProjectPalettes(id))
        }catch(error){

        }
    }
}