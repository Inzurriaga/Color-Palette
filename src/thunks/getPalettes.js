import { initialPalette } from "../action"
export const getPalettes = () => {
    return async (dispatch) => {
        try{
            const response = await fetch("http://localhost:3001/api/palettes")
            if(!response.ok) {
                throw Error(response.statusText)
              }
            const palettes = await response.json()
            dispatch(initialPalette(palettes))
        }catch(error){

        }
    }
}