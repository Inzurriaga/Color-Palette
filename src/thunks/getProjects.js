import { initialProjects } from "../action"

export const getProjects = () => {
    return async (dispatch) => {
        try{
            const response = await fetch("http://localhost:3001/api/projects")
            if(!response.ok) {
                throw Error(response.statusText)
              }
            const projects = await response.json()
            dispatch(initialProjects(projects))
        }catch(error){

        }
    }
}