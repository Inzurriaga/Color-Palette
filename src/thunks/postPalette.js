import { addNewPalette } from "../action"

export const postPalette = (id, palette) => {
    const method = {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify(palette)
    }
    return async (dispatch) => {
        try{
            const response = await fetch(`http://localhost:3001/api/projects/palettes/${id}`, method);
            if(!response.ok) {
                throw Error(response.statusText)
              }
            const paletteId = await response.json();
            dispatch(addNewPalette({...palette, id: paletteId.id, "project_id": id}))
        }catch(error){
            console.log(error)
        }
    }
}