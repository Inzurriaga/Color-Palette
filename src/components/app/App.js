import React, { Component } from 'react';
import { connect } from "react-redux"
import Header from "../header/Header.js"
import ColorPalette from "../colorPalette/ColorPalette.js"
import ProjectDisplay from "../projectDisplay/ProjectDisplay.js"
import SaveModal from "../modal/SaveModal"
import { getProjects } from "../../thunks/getProjects";
import { getPalettes } from "../../thunks/getPalettes";
import { generateInitialColorPalette } from "../../action"

export class App extends Component {

  componentDidMount(){
    this.props.getProjects()
    this.props.getPalettes()
    this.props.generateInitialColorPalette()
  }

  render(){
    return (
      <div className="App">
        <Header/>
        <ColorPalette/>
        <ProjectDisplay/>
        {
            this.props.toggleSaveModal && <SaveModal />
        }
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  getProjects: () => dispatch(getProjects()),
  getPalettes: () => dispatch(getPalettes()),
  generateInitialColorPalette: () => dispatch(generateInitialColorPalette())
})

export const mapStateToProps = (state) => ({
  toggleSaveModal: state.toggleSaveModal,
}) 

export default connect(mapStateToProps, mapDispatchToProps)(App);
