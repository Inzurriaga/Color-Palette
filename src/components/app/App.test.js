import React from "react"
import { shallow } from "enzyme"
import { App, mapDispatchToProps, mapStateToProps } from "./App"
import { getProjects } from "../../thunks/getProjects";
import { getPalettes } from "../../thunks/getPalettes";
import { generateInitialColorPalette } from "../../action"
jest.mock("../../thunks/getProjects")
jest.mock("../../thunks/getPalettes")

const mockGetProjects = jest.fn()
const mockGetPalettes = jest.fn()
const mockGenerateInitialColorPalette = jest.fn()
const mockToggleSaveModal = false

describe("App", () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <App getProjects={mockGetProjects}
                getPalettes={mockGetPalettes}
                generateInitialColorPalette={mockGenerateInitialColorPalette}
                toggleSaveModal={mockToggleSaveModal}
                />
        )
    })

    describe("App", () => {

        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })

    })

    describe("mapDispatchToProps", () => {
        
        it('calls dispatch with an generateInitialColorPalette action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = generateInitialColorPalette()
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.generateInitialColorPalette()
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })

          it('calls dispatch with an getProjects action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = getProjects()
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.getProjects()
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })

          it('calls dispatch with an getPalettes action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = getPalettes()
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.getPalettes()
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })

    })
    describe("mapStateToProps", () => {

        it("should only return waht is needed from the store", () => {
            const mockStore = {
                colorPalette: [{},{},{}],
                project: [{},{},{}],
                toggleSaveModal: false
            }
            const expected = {
                toggleSaveModal: false
            }
            const result = mapStateToProps(mockStore)
            expect(result).toEqual(expected)
        })
        
    })
})