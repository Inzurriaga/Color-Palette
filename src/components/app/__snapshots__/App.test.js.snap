// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snap shot 1`] = `
<div
  className="App"
>
  <ConnectFunction />
  <ConnectFunction />
  <ConnectFunction />
</div>
`;
